pkg/metrics/metrics.go

```diff
@@ -31,10 +36,30 @@ var UploadRejectionsCounter = prometheus.NewCounterVec(
 	[]string{"operation", "reason"},
 )
 
-func RegisterCommonMetrics(registerer prometheus.Registerer) error {
-	if err := registerer.Register(UploadRejectionsCounter); err != nil {
-		return fmt.Errorf("failed to register rejected uploads count metric: %w", err)
-	}
+var RemoteSecretConditionGauge = prometheus.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Namespace: config.MetricsNamespace,
+		Subsystem: config.MetricsSubsystem,
+		Name:      "status_condition",
+		Help:      "The status condition of a specific RemoteSecret",
+	},
+	[]string{"name", "namespace", "condition", "status"},
+)
 
+func RegisterCommonMetrics(registerer prometheus.Registerer) error {
+	registerer.MustRegister(UploadRejectionsCounter, RemoteSecretConditionGauge)
 	return nil
 }
+
+func DeleteRemoteSecretCondition(ctx context.Context, name, namespace string) {
```

Comment:
imo those 2 functions are too specific to be placed there, but i understood you don't want to pollute controller with those utility func-s.